### PR TITLE
Merge workflows  | Add release workflow | Windows Compatible npm test script

### PR DESCRIPTION
**Pull request by Madoshakalaka**
Now GitHub actions also releases zipped chrome extension. It's the format chrome extension store expects. And also one can install click-in-text privately with the zipped file.

Also our three CI/CD tasks (tests, lint-commit, release) are now in one job. I thought thoroughly about this: multiple concurrent workflows or multiple jobs inside one workflow may sound nice. But we don't need them. They just duplicate actions run time in our case. That's `actions/checkout` `actions/setup-node` `npm ci` and `npm build` in the release case.

see more in the comments documented in the yaml files.

And sorry about the "windows-compatible npm test script" part - I meant to separate into two PRs but submitted together accidentally. But they are small commits and it should be too much a problem?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
     "build": "cross-env NODE_ENV=production webpack && node libexec/fix-non-ascii-contentscript.js && node libexec/zip-extension.js",
     "build_dist": "cross-env NODE_ENV=production webpack",
     "dev": "webpack --watch",
-    "test": "npx http-server dist & npx wait-on http-get://localhost:8080 && npx cypress run --spec 'cypress/integration/*'",
```
Comment (Madoshakalaka) [inline package.json:6]: Windows cmd does not understand ampersand (running in background)

File: libexec/integration-test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const waitOnProcess = spawn('npx', ['wait-on', 'http-get://localhost:8080/test.html'], {shell:true})
+
+waitOnProcess.on('close', () => {
+  console.log('starting cypress tests')
+  
+  const cypressProcess = spawn('npx', ['cypress', 'run', '--spec', 'cypress/integration/*'], {shell:true})
+  cypressProcess.stdout.on('data', data=>console.log(data.toString()))
+  cypressProcess.stderr.on('data', data=>console.error(data.toString()))
+  cypressProcess.on('close', code=>process.exit(code))
+})
```
Comment (Madoshakalaka) [inline libexec/integration-test.js:16]: This is old `npm test` rewritten in javascript.

File: .github/workflows/integration-lint-release.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          draft: false
+          prerelease: false
+
+        - name: Upload Release Asset
+          if: github.event_name == 'push'
+          uses: actions/upload-release-asset@v1
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          with:
+            upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
```
Comment (eddieantonio) [inline .github/workflows/integration-lint-release.yml:79]:
```suggestion
            upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing its ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
```

File: libexec/integration-test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @file This script starts dev server and runs cypress tests.
+ */
+
+const { spawn } = require('child_process')
+spawn('npx', ['http-server', 'dist'],{shell:true})
```
Comment (eddieantonio) [inline libexec/integration-test.js:6]: Minor style suggestion — my mind is so used to separating imports from code that uses imports, that adding a single space makes this much easier for me to scan! 😅 

```suggestion
const { spawn } = require('child_process')

spawn('npx', ['http-server', 'dist'], {shell: true})
```

File: libexec/integration-test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+const waitOnProcess = spawn('npx', ['wait-on', 'http-get://localhost:8080/test.html'], {shell:true})
+
+waitOnProcess.on('close', () => {
+  console.log('starting cypress tests')
+  
+  const cypressProcess = spawn('npx', ['cypress', 'run', '--spec', 'cypress/integration/*'], {shell:true})
+  cypressProcess.stdout.on('data', data=>console.log(data.toString()))
+  cypressProcess.stderr.on('data', data=>console.error(data.toString()))
```
Comment (eddieantonio) [inline libexec/integration-test.js:14]: I think there's a  way to call `spawn` such stdout/stderr are already piped to the correct places...  See here: https://nodejs.org/api/child_process.html#child_process_options_stdio

**Review by eddieantonio — APPROVED** [review]
LGTM! I think the new, single file workflow is rather clean!

File: libexec/integration-test.js (hunk 5 above)
Comment (Madoshakalaka) [reply libexec/integration-test.js:15]: thank you! Was wondering too!

**Pull request merged**